torch/csrc/jit/ir.h

```diff
@@ -1092,6 +1092,23 @@ friend struct Block;
     result->output()->setType(FloatType::get());
     return result;
   }
+  Node* createTensorAttr(Value* value, const std::string& field) {
+    JIT_ASSERT(*value->type() == *DynamicType::get());
+    if (field == "dtype") {
+      auto* result = create(prim::TensorDType, {value});
+      result->output()->setType(IntType::get());
+      return result;
+    } else if (field == "device") {
+      auto* result = create(prim::TensorDevice, {value});
+      result->output()->setType(ListType::create(IntType::get()));
+      return result;
+    } else if (field == "shape") {
+      auto* result = create(prim::TensorShape, {value});
+      result->output()->setType(ListType::create(IntType::get()));
+      return result;
+    }
+    return nullptr;
+  }
```

Comment:
I don't think this should live in `ir.h`. It's only used in one single place in the front end, so there's no need to put this in this huge file.